handshake/connection_parameters_manager.go

```diff
@@ -61,14 +66,36 @@ var (
 )
 
 // NewConnectionParamatersManager creates a new connection parameters manager
-func NewConnectionParamatersManager(pers protocol.Perspective, v protocol.VersionNumber) ConnectionParametersManager {
+func NewConnectionParamatersManager(
+	pers protocol.Perspective, v protocol.VersionNumber,
+	maxReceiveStreamFlowControlWindowServer protocol.ByteCount, maxReceiveConnectionFlowControlWindowServer protocol.ByteCount,
+	maxReceiveStreamFlowControlWindowClient protocol.ByteCount, maxReceiveConnectionFlowControlWindowClient protocol.ByteCount,
+) ConnectionParametersManager {
+	if maxReceiveStreamFlowControlWindowServer == 0 {
+		maxReceiveStreamFlowControlWindowServer = protocol.DefaultMaxReceiveStreamFlowControlWindowServer
+	}
+	if maxReceiveConnectionFlowControlWindowServer == 0 {
+		maxReceiveConnectionFlowControlWindowServer = protocol.DefaultMaxReceiveConnectionFlowControlWindowServer
+	}
+	if maxReceiveStreamFlowControlWindowClient == 0 {
+		maxReceiveStreamFlowControlWindowClient = protocol.DefaultMaxReceiveStreamFlowControlWindowClient
+	}
+	if maxReceiveConnectionFlowControlWindowClient == 0 {
+		maxReceiveConnectionFlowControlWindowClient = protocol.DefaultMaxReceiveConnectionFlowControlWindowClient
+	}
+
```

Comment:
Use the `populateServerConfig` and `populateClientConfig` to set default values. The idea here is that once the `quic.Config` has passed through one of these functions (which happens right after the `quic.Config` is passed to quic-go), all values in the config are valid.
This will also allow you to just pass 2 function parameters here.